### PR TITLE
Add initial test stubs for Remote Execution

### PR DESCRIPTION
**Pull request by JacobCallahan**
The Remote Execution (Management) feature is currently confined to UI.
These initial test stubs are the current manual testing cases.

### CODE REVIEW

File: tests/foreman/ui/test_remoteexecution.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
+"""Test class for Remote Execution Management UI"""
+from robottelo.decorators import run_only_on, stubbed
+from robottelo.test import UITestCase
+
+
+class TestRemoteExecution(UITestCase):
```
Comment (oshtaier) [inline tests/foreman/ui/test_remoteexecution.py]: `TestRemoteExecution` -> `RemoteExecutionTestCase`

**Comment by oshtaier** [comment]
Please refer to https://github.com/SatelliteQE/robottelo/issues/3083 for our naming conventions.
All test cases should use words `positive` or `negative` after word `test_`

**Comment by oshtaier** [comment]
Don't need to use extra line in the end of docstring. Change
```
 +        @Status: Manual
 +
 +        """
```
on
```
 +        @Status: Manual
 +        """
```

**Comment by oshtaier** [comment]
Please classify all your test cases using corresponding `tier` decorators as in all our tests

**Comment by oshtaier** [comment]
Please, don't use numbers in test names (e.g. `test_re_create_job_template_1`) and provide meaningful name instead

**Comment by oshtaier** [comment]
Pending comments

**Comment by JacobCallahan** [comment]
Thanks for the great input, @oshtaier!

File: tests/foreman/ui/test_remoteexecution.py [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    """Test class for remote execution feature"""
+
+    @stubbed()
+    @run_only_on('sat')
+    def test_re_create_job_template_1(self):
```
Comment (sthirugn) [inline tests/foreman/ui/test_remoteexecution.py]: remove `re_` part from all test cases as all the test cases in this module belong to re - remote execution.

Comment (sthirugn) [inline tests/foreman/ui/test_remoteexecution.py]: Also tests need to have `positive`, `negative` text added in the test case names whenever possible.

**Comment by JacobCallahan** [comment]
Addressed all previous comments. I'm not 100% on the tiers, but these should be about right.

File: tests/foreman/ui/test_remoteexecution.py [hunk 3]
```diff
@@ -0,0 +1,227 @@
+"""Test class for Remote Execution Management UI"""
+from robottelo.decorators import stubbed, tier1, tier2
+from robottelo.test import UITestCase
+
+
+class TestRemoteExecution(UITestCase):
```
Comment (oshtaier) [inline tests/foreman/ui/test_remoteexecution.py]: Not sure what is changed, but I still see `TestRemoteExecution` title when it should be like I mentioned in my previous comment:
`TestRemoteExecution` -> `RemoteExecutionTestCase`

File: tests/foreman/ui/test_remoteexecution.py [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+        6. Click submit
+
+        @Assert: The job template was successfully created
+
+        @Status: Manual
+        """
+
+    @stubbed()
+    @tier1
+    def test_positive_create_job_input_template(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_remoteexecution.py]: It seems more like `test_positive_create_job_template_input`, but not sure on 100%

File: tests/foreman/ui/test_remoteexecution.py [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+        9. Click submit
+
+        @Assert: The job template was successfully created
+
+        @Status: Manual
+        """
+
+    @stubbed()
+    @tier1
+    def test_negative_create_job_template(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_remoteexecution.py]: `test_negative_create_job_template_with_same_name` is more applicable here.
I suggest also adding test case that targets invalid names for job template (e.g. too long strings used for name).

**Comment by oshtaier** [comment]
Can we add few test cases for update or delete procedures for mentioned `job templates`? I didn't see UI for that functionality, so please sorry in advance for incorrect suggestions

**Comment by oshtaier** [comment]
Also, please add corresponding entry to robottelo/docs/api/tests.foreman.ui.rst

**Comment by sghai** [comment]
@JacobCallahan : hey..good to see tests around remote execution.  Are you planning to cover more scenarios around this later ?

File: tests/foreman/ui/test_remoteexecution.py [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+        """Create Job Template with duplicate name
+
+        @Feature: Remote Execution
+
+        @Steps:
+
+        1. Create a valid job template
+        2. Enter a name that has already been used
```
Comment (rplevka) [inline tests/foreman/ui/test_remoteexecution.py]: You might want to add a step of pre-creating the 1st job template first. You can use the @Setup section of the docstring for such steps

File: tests/foreman/ui/test_remoteexecution.py [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+        4. Select "preview" within the template viewer
+
+        @Assert: Verify no errors are thrown
+
+        @Status: Manual
+        """
+
+    @stubbed()
+    @tier1
+    def test_positive_preview_verify_file(self):
```
Comment (rplevka) [inline tests/foreman/ui/test_remoteexecution.py]: not sure about this one but shouldn't this be a negative scenario?

Comment (rplevka) [inline tests/foreman/ui/test_remoteexecution.py]: negative?

Comment (oshtaier) [inline tests/foreman/ui/test_remoteexecution.py]: +1

File: tests/foreman/ui/test_remoteexecution.py (hunk 6 above)
Comment (sthirugn) [inline tests/foreman/ui/test_remoteexecution.py]: :+1:

File: tests/foreman/ui/test_remoteexecution.py [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+        1. Create a working job template
+        2. Go to an individual host and click Run Job
+        3. Select the job and appropriate template
+        4. Select "Schedule Future Job"
+        5. Enter a desired time for the job to run
+        6. Click submit
+
+        @Assert: Verify the job was not immediately ran
+        @Assert: Verify the job was succesfully ran after the designated time
+
```
Comment (sthirugn) [inline tests/foreman/ui/test_remoteexecution.py]: Only one `@Assert` section allowed.  So probably you can number these different assertions under `@Assert` section like how you did for `@Steps`

**Comment by sthirugn** [comment]
Pending @rplevka's comment about using `@setup` section for pre-requisites.

File: docs/api/tests.foreman.ui.rst [hunk 9]
```diff
@@ -198,6 +198,11 @@
 
 .. automodule:: tests.foreman.ui.test_puppetclass
 
+:mod:`tests.foreman.ui.test_remoteexecution`
+---------------------------------------
```
Comment (oshtaier) [inline docs/api/tests.foreman.ui.rst]: few more symbols to avoid new warning
```
:mod:`tests.foreman.ui.test_remoteexecution`
--------------------------------------------
```

File: tests/foreman/ui/test_remoteexecution.py (hunk 8 above)
Comment (oshtaier) [inline tests/foreman/ui/test_remoteexecution.py]: +1

**Comment by oshtaier** [comment]
ACK
pending my small comment and @rplevka ones

**Comment by rplevka** [comment]
ACK
pending comments :)

**Comment by JacobCallahan** [comment]
I believe I have addressed all previous comments. Thanks for the great input, guys. A number of things have changed since my last PR!! 

I also cleaned up a few other lines, per what @oshtaier recommended.

**Comment by sthirugn** [comment]
ACK

**Comment by rplevka** [comment]
gg,
ACK

**Comment by omaciel** [comment]
@JacobCallahan :100:
